compiler/psi/src/org/jetbrains/kotlin/parsing/KotlinExpressionParsing.java

```diff
@@ -1836,8 +1841,17 @@ public void parseValueArgumentList() {
                 while (true) {
                     while (at(COMMA)) errorAndAdvance("Expecting an argument");
                     parseValueArgument();
-                    if (at(COLON) && lookahead(1) == IDENTIFIER) {
-                        errorAndAdvance("Unexpected type specification", 2);
+                    if (at(COLON) && (lookahead(1) == IDENTIFIER || lookahead(1) == LPAR)) {
```

Comment:
I suggest removing checking `LPAR`: in `foo(x:)`,`Unexpected type specification` error may be confused because in fact, there isn't any type here.